web/src/main/java/com/epam/esm/controller/GiftCertificateController.java

```diff
@@ -15,40 +16,41 @@
 @RestController
 @RequestMapping("/certificates")
 public class GiftCertificateController {
+
     private static final Logger logger = LogManager.getLogger(GiftCertificateController.class);
     private CertificateService certificateService;
 
+    @Autowired
+    private CertificateHateoasBuilder certificateHateoasBuilder;
+
     @Autowired
     public GiftCertificateController(CertificateService certificateService) {
         this.certificateService = certificateService;
     }
 
     /**
-     * Extracts MultiValueMap to get proper parameters for search query
+     * Extracts search query to get a list of certificates
      * @param
      * @return List of Certificate type
      */
 
     @RequestMapping(method = RequestMethod.GET)
-    public List<Certificate> getByParameters(@RequestParam(required = false) MultiValueMap<String, String> params) {
-        try {
-            return certificateService.getByParameters(params);
-        } catch (ServiceException e) {
-            logger.error(e);
-            throw e;
-        }
+    public List<CertificateDto> getByParameters(SearchQuery searchQuery) {
+        List<CertificateDto> list = certificateService.getByParameters(searchQuery);
+        certificateHateoasBuilder.buildToEntitiesCollection(list);
+        return list;
     }
 
     /**
      * Adds new Certificate to DB
-     * @param certificate full certificate entity
+     * @param certificateDto full certificate entity
      * @return Certificate object
      */
 
     @RequestMapping(method = RequestMethod.POST)
-    public Certificate create(@RequestBody @Validated Certificate certificate) {
+    public CertificateDto create(@RequestBody @Validated CertificateDto certificateDto) {
         try {
-            return certificateService.add(certificate);
+            return certificateService.add(certificateDto);
```

Comment:
Why no hateoas for post/put/delete operations? where should client go after these operations?

Reply:
Added hateoas for post/put operations. My delete operations are void-type